\documentclass[a4paper]{article}
\usepackage{graphicx}
\usepackage{amsmath}
\usepackage{xr}
\usepackage{subcaption}
\usepackage{authblk}

\usepackage[round]{natbib}
\bibliographystyle{apalike}

\usepackage{lineno}
\usepackage{xcolor}
\usepackage{setspace}

\externaldocument[supp-]{Appendix}

\allowdisplaybreaks

\doublespacing

\title{Process-based modelling of microbial community dynamics in the human colon}

\author[1,*]{Helen Kettle}
\author[2]{Petra Louis}
\author[2]{Harry J. Flint}
\affil[1]{Biomathematics and Statistics Scotland, James Clerk Maxwell Building, Peter Guthrie Tait Road, Edinburgh, EH9 3FD}
\affil[2]{Gut Health Group, Rowett Institute, University of Aberdeen, Aberdeen, UK}
\affil[*]{Corresponding author: Helen.Kettle@bioss.ac.uk}


\begin{document}

\maketitle

\linenumbers


\begin{abstract}
The human colon contains a dynamic microbial community whose composition has important implications for human health. In this work we build a process-based model of the colonic microbial ecosystem and compare with general empirical observations and the results of in-vivo experiments.   
Our model comprises a complex microbial ecosystem along with absorption of short chain fatty acids (SCFA) and water by the host through the gut wall, variations in incoming dietary substrates (in the form of ``meals" whose composition varies in time), bowel movements, feedback on microbial growth from changes in pH resulting from SCFA production, and multiple compartments to represent the proximal, transverse and distal colon. 
We verify our model against a number of observed criteria, e.g. total SCFA concentrations, SCFA ratios, mass of bowel movements, pH and water absorption over the transit time; and then run simulations investigating the effect of colonic transit time, and the composition and amount of indigestible carbohydrate in the host diet, which we compare with in-vivo studies.
The code is available as an R package (microPopGut) to aid future research.
\end{abstract}


\section*{Introduction}
The human colon harbours a dense and diverse community of microbiota whose interactions with the host can have a profound effect on human health (e.g. \cite{health1}, \cite{morrison2016}). Due to the location of this community within its host, data collection and experimentation are problematic. Information on this system mostly comes from volunteer experiments in which diet and stool samples are monitored or from laboratory experiments using the microbes found in stool samples. Another approach is to put current knowledge into a mathematical framework and run simulations of the system to test our understanding and identify knowledge gaps. To this end a number of mathematical models of this system have been developed - e.g. 
\cite{Cremer16}, \cite{Cremer17}, \cite{Munoz}, \cite{smith}, \cite{moorthy}.

When developing a model, a number of assumptions about the system are made in order to reduce complexity/dimensionality so that the model is easier to parameterise, run and analyse. Some modellers choose to reduce the microbial complexity and focus on the physics of the gut (e.g. \cite{Cremer16}, \cite{Cremer17}), some try to achieve a balance of both (e.g. \cite{Munoz}) and some choose to develop the microbial community (e.g. \cite{smith}). The model described here focuses on the microbial community dynamics and on interactions with the host, with a fairly simple model of the colon. We include the simulation of `meals' (of random composition and size) arriving at the colon and look at the effects of bowel movements, both of which, as far as we are aware, have not been previously incorporated into such models.
Having developed a complex model of human gut microbiota in a fermentor system \citep{Kettle2015}, and publicly available software (microPop - an R package for modelling microbial communities \citep{Kettle2018}) we now incorporate this 10-group microbial ecosystem model (Table \ref{table:groups}) into a model of the human gut in order to simulate the effects of diet and host on the microbial composition and subsequent short chain fatty acid (SCFA) production. 

Since approximately 95\% of the SCFA produced by the microbes during growth are absorbed by the host through the gut wall this represents a strong interaction between the microbes and the host. 
Indeed the ratio of the 3 main SCFAs (acetate, butyrate and propionate) is known to have a significant effect on human health (\cite{louis2014}, \cite{morrison2016}).
Thus, we prioritise information on the values of these ratios in our model verification.
Similarly approximately 90\% of the water flowing into the colon is absorbed. Changes in the volume of water have a significant effect on the concentration of the molecules in the colon which in turn affects pH which then affects microbial growth, all of which are included in our model.

Due to its shape within the body, the colon is commonly divided into 3 different regions - the proximal, transverse and distal sections running from beginning to end (Fig. \ref{fig:Validation}A and B). 
The availability of substrate, microbial growth and hence pH vary along the colon, therefore, although our model is not spatial we simulate these three regions explicitly, with flow from one to another.
Furthermore, as well as incorporating varying substrate inflow in the form of meals we also add in the release of mucins along the length of the colon which can be microbially broken down to release proteins and carbohydrates, allowing for further microbial growth away from the beginning of the colon where the substrates enter.
A graphical summary of the model is shown in Fig. \ref{fig:schematic}, the microbial functional groups are shown in Table \ref{table:groups} and the model state variables are summarised in Table \ref{table:stateVariables}.

We use the following criteria to verify our model captures the main features established for the system:
\begin{enumerate}
\item Total SCFA (TSCFA) concentration in the proximal, transverse and distal compartments should be around 123, 117 and 80 mM respectively according to sudden death human autopsies \citep{cummings} 
\item Acetate:Propionate:Butyrate ratios are similar (around 3:1:1) in all regions of the colon and around 60:20:20 mM \citep{cummings}
\item Over 95\% of SCFA are absorbed by the host \citep{Topping}
\item Approx. 90\% of incoming water is absorbed by the host \citep{Phillips} 
\item pH in the proximal, transverse and distal compartments should be around 5.7, 6.2 and 6.6 respectively (\cite{cummings}, and telemetry data from \cite{miko}, \cite{bown1974})
\item Normal daily fecal output in Britain is 100-200 g d$^{-1}$ of which 25-50 g is solid matter (i.e. 50-175 g d$^{-1}$ is water). Bacteria make up about 55\% of the solid matter i.e. 14-28 g d$^{-1}$ of microbes emitted \citep{StephenCummings}.
\item TSCFA concentration decreases with transit time \citep{Lewis}
\end{enumerate}
After model verification we examine the effects of including meals, bowel movements and fixed/varying pH into the model. We then use the model to look at how carbohydrate composition (based on the fractions of resistant starch (RS) and non-starch polysaccharides (NSP)) and total carbohydrate affect the microbial community and SCFA composition. The simulations are then compared with in-vivo data from human volunteer experiments. 

Although gut microbiota are highly complex and not fully understood, here we show that it is nonetheless possible to develop predictive models of key components of this ecological system. 
An important goal of our modelling is to aid and inform the interpretation of data obtained, mostly from faecal samples, in studies on diet and health in humans.
Our results show promise and we believe this model represents a significant step forward in analysing this highly complex system. 
We refer to the model as ``microPopGut" and to aid future research the code is available as an R-package on github (https://github.com/HelenKettle/microPopGutCode) and instructions on how to use the package are given in the supplementary file `getStartedWithMicroPopGut.pdf'.


\begin{figure}
    \centering
      \includegraphics[scale=0.35]{images/Fig1Schematic.pdf}
    \caption{A. Colon schematic, B. Table of typical values for physical properties (length, volume, pH and TSCFA) and C. plots of summarised model simulations for average TSCFA and pH for comparison with typical values. Red dots show results from meals-inflow averaged over 4 random seeds; blue dots show results from continuous substrate inflow. 
    }
    \label{fig:Validation}
\end{figure}



\begin{table}[ht]
    \caption{Microbial functional groups included in the model (and the R package microPop \citep{Kettle2018}) and described by \cite{Kettle2015}. Users should be aware that the parameter values given in the data frames in the software will almost certainly change with increasing knowledge of gut microbiota and in some cases are simply a ``best guess''. }
    \small
    \centering
\begin{tabular}{|p{0.24\linewidth} | p{0.06\linewidth} | p{0.3\linewidth} | p{0.4\linewidth}|}
        \hline
microPop Name & Abbr. &	Description &	Examples\\
\hline
Bacteroides 	& B &Acetate-propionate-succinate group & 	\textit{Bacteroides} spp., \textit{Prevotella} spp.,  \textit{Akkermansia muciniphila} (Verrucomicrobia)\\
NoButyStarchDeg & NBSD &Non-butyrate-forming starch degraders & \textit{Ruminococcaceae} related to \textit{Ruminococcus bromii}. Also includes certain \textit{Lachnospiraceae}\\
NoButyFibreDeg & NBFD &	Non-butyrate-forming fibre degraders &	\textit{Ruminococcaceae} related to \textit{Ruminococcus albus}, \textit{Ruminococcus flavefaciens}. Also includes certain \textit{Lachnospiraceae}\\
LactateProducers & LP&	Lactate producers &	\textit{Actinobacteria}, especially \textit{Bifidobacterium} spp., \textit{Collinsella aerofaciens}\\
ButyrateProducers1  & BP1& 	Butyrate Producers &	\textit{Lachnospiraceae} related to \textit{Eubacterium rectale}, \textit{Roseburia} spp.\\
ButyrateProducers2 & BP2 &	Butyrate Producers &	Certain \textit{Ruminococcaceae}, in particular \textit{Faecalibacterium prausnitzii}\\
PropionateProducers & PP&	Propionate producers &	\textit{Veillonellaceae} e.g. \textit{Veillonella} spp., \textit{Megasphaera elsdenii}\\
ButyrateProducers3 & BP3&	Butyrate Producers & \textit{Lachnospiraceae} related to \textit{Eubacterium hallii}, \textit{Anaerostipes} spp.\\
Acetogens & A &	Acetate Producers &	Certain \textit{Lachnospiraceae}, e.g. \textit{Blautia hydrogenotrophica}\\
Methanogens & M&	Methanogenic archaea &	\textit{Methanobrevibacter smithii}\\
   \hline 
    \end{tabular}

    \label{table:groups}
\end{table}   
 



\begin{table}[ht]
    \caption{State variables included in the model. They are all in units of mass (g; with the exception of pH) and they are computed for each model compartment (e.g. prox., trans. and dist.). They are derived automatically from the substrates and metabolites specified for each microbial functional group (MFG) in the input file/dataframe to the R package microPop \cite{Kettle2018}.}
\begin{tabular}{|l|l|l|}
\hline
Name &  Details\\
\hline
Microbial biomass & Computed for each of the 10  MFGs (Table \ref{table:groups})\\
\hline
water  & from dietary intake or from microbial metabolism \\
Protein  & from dietary intake or mucin\\
Resistant starch (RS)  & from dietary intake or mucin\\
Non-starch polysaccharides (NSP)  & from intake or mucin\\
\hline
pH & Computed from TSCFA (Eq. \ref{eq:pH(TSCFA)})\\
\hline
Acetate &   metabolite\\
Butyrate &   metabolite\\
Propionate  &  metabolite\\
Formate &  metabolite \\
Carbon dioxide &  metabolite \\
Methane &  metabolite \\
Ethanol &  metabolite \\
Lactate &  metabolite \\
Succinate &  metabolite\\
Hydrogen &  metabolite\\
\hline
\end{tabular}
\label{table:stateVariables}
\end{table}


\begin{figure}
    \centering
   \includegraphics[scale=0.3]{images/microbeEco4.pdf}
   \caption{Model system with the microbial ecosystem comprising 10 microbial functional groups (Table \ref{table:groups}) which consume substrates (RS, NSP and protein) and water. The microbes produce metabolites some of which are consumed by other MFGs (`cross-feeding'). SCFA and water are absorbed through the colon wall (at a different specific rates). The system shown within the dashed line is repeated in each of the modelled regions of the colon (proximal, transerve, and distal) with the contents of the previous region, flowing into the next. The first compartment (proximal) has inflow from the small intestine - this can be constant inflow or simulated meals whose composition varies randomly in time. The third model compartment (distal) has outflow to stool which can be constant or evacuation via bowel movements can be simulated. pH varies with the TSCFA concentration and affects the rate of microbial growth differently for each MFG.
    }
    \label{fig:schematic}
\end{figure}



\section*{Results} 

\subsection*{Standard Model}
The model settings which give the best fit to our criteria are shown in Table \ref{table:drivers} (colon parameters and dietary inflow). The microbial group parameters are listed in Supp. Info. (section 3). These define our default model. From this we investigate the effects of different model configurations, e.g. with/without bowel movements, meals and variable pH, for a range of transit times.
Simulations with meals have a random component therefore the model is run for a number of different starting seed values.
Due to the random fluctuations these simulations will not reach steady state therefore the summary values are taken as the mean from day 7 (to remove the effect of the initial conditions) to the end of the simulation (28 days) and are averaged over multiple seeds.

Table \ref{table:Results} gives summary results of the model simulations without bowel movements but with varying pH for each bowel region. Fig. \ref{fig:std} shows results from more simulations but for the distal colon only.
Fig. \ref{fig:std}a shows that although bowel movements make a difference to the total biomass and the TSCFA they do not have a large effect on the community composition or the SCFA ratios. Thus in the interests of model simplicity we decide to not include bowel movements in later simulations.
However, varying pH with TSCFA can be seen to make a large difference to the microbial community (Fig. \ref{fig:std}b) and also improves the SCFA ratios with respect to our verification criteria.
The addition of meals makes a significant difference which increases with increasing transit time (Fig. \ref{fig:std}c). 
In Fig. \ref{fig:flucInflowTt} the time series output from the model shows how the meals-inflow allows the community to experience large shifts over time (on a much longer time scale than the variations in the input), as opposed to the fixed state approached using a constant substrate inflow.

Fig. \ref{fig:Validation}C shows the average pH and TSCFA for the proximal, transverse and distal compartments. 
A decrease in TSCFA (and concomitant increase in pH) with longer transit time is predicted in the proximal colon both for meal inflow and continuous input and this is in broad agreement with experimental findings (Lewis and Heaton 1997). In section 2 of the Supp. Info. we suggest a mathematical explanation for this based on the supposition that the specific rate of absorption of water through the gut wall is slower than that for SCFA.


Regarding Table \ref{table:Results}, for some criteria, e.g. pH, the continuous inflow setting gives results closer to our verification values, but in other cases, e.g. A:B:P in distal colon, simulating meals gives closer results. 
Note that we consider a transit time of 1 day the most typical of the three transit times, and the one that should be compared with our verification criteria, the others are included to show the variation in results.
Ideally TSCFA should be 123, 117 and 80 mM for prox., trans., dist. but the best match we have to this is for a 3 d transit time and continuous inflow. 
This is most likely due to the fact that our model has fixed rates of specific absorption of SCFA and water throughout the colon.
However, our TSCFA values are within a reasonable range and display the general trend of decreasing TSCFA from the proximal to distal colon.
The microbe output, i.e. the outflow of fecal microbes is steady at around 20 g d$^{-1}$ in all cases which fits well in the verification range (14-28 g d$^{-1}$).
The water fraction is the ratio of the rate of fecal water over the rate of water flowing into the colon, since 90\% of water is absorbed this should be 0.1. This is approximately correct for our 1 d simulations (0.14) but, as expected, when transit time increases this decreases significantly.
In summary, comparing these simulation results with our list of model verification criteria shows that in general our model is fit for purpose, and that the inclusion of meals-inflow and varying pH improve our simulations. 


\begin{table}[ht]
\caption{Summary of model results (for comparison with our list of criteria) for 3 different transit times, with meals or continuous inflow and with pH varying with TSCFA. Microbe output is the mass of microbes leaving the colon per day and the water fraction is amount of water leaving the colon per day divided by the amount entering. All simulations were run for 28 days and the results shown are the average over days 7-28. The results for the simulations with meals are averaged over 4 random seeds. `A:B:P dist' refers to the Acetate:Butyrate:Propionate ratio (mM) in the distal colon.}
\centering
\small
\begin{tabular}{l|rrr|rrr}
  \hline
  & \multicolumn{3}{c|}{Meals}  & \multicolumn{3}{c}{Continuous  inflow} \\
\hline
transit  time & 1d & 2d & 3d & 1d & 2d & 3d \\ 
  \hline
TSCFA prox  (mM) & 115.3 & 110.3 & 105.2 & 124.4 & 123.5 & 122.1 \\ 
TSCFA trans (mM) & 102.1 & 88.1 & 83.4 & 75.5 & 96.1 & 111.1 \\ 
TSCFA  dist (mM) & 107.6 & 64.8 & 69.0 & 89.5 & 62.0 & 83.3 \\ 
A:B:P dist (mM) & 62:28:17 &31:23:10 & 34:23:12 & 56:27:7 &38:18:6 &59:17:7\\
\hline
pH prox & 6.0& 6.1 & 6.2 & 5.9 & 5.9 & 5.9 \\ 
pH trans & 6.2 & 6.5 & 6.6 & 6.7 & 6.4 & 6.1 \\ 
pH dist & 6.2 & 6.9 & 6.8 & 6.5 & 6.9 & 6.6 \\ 
\hline
microbe  output (g d$^{-1}$) & 20.2 & 20.1 & 20.1 & 20.0 & 20.1 & 20.0 \\ 
water  fraction & 0.14 & 0.04 & 0.02 & 0.14 & 0.04 & 0.02 \\ 
   \hline
\end{tabular}
\label{table:Results}
\end{table}


 \begin{figure}
    \centering
    \begin{subfigure}[t]{0.32\textwidth}%
      \caption{Bowel Movements}%
      \includegraphics[scale=0.35]{images/BMpdbarSmall.eps}
    \end{subfigure}
    \begin{subfigure}[t]{0.32\textwidth}
      \caption{pH}
      \includegraphics[scale=0.35]{images/fixpHbarSmall.eps}
    \end{subfigure}
    \begin{subfigure}[t]{0.32\textwidth}
      \caption{Transit time}
      \includegraphics[scale=0.35]{images/transitTimebarSmall.eps}
    \end{subfigure}
    \caption{Summary results (averaged over days 7-28 and over random seeds) for the distal compartment for continuous inflow or fluctuating inflow (i.e. `meals') for continuous outflow from colon or for 2 bowel movements per day (`2 BM/d'). The RS fraction is 0.78 (i.e. 78\% of the dietary carbohyrate is resistant starch and 22\% is NSP) and the transit time is 0.93 d for a), 1.25 d for b) and at 0.25, 0.5, 1, 1.5, 2, 2.5 and 3 days for c). The top row shows the biomass of each group, the bottom row shows the SCFA.}
    \label{fig:std}
\end{figure}


 \begin{figure}
    \centering
   \includegraphics[scale=0.5]{images/transitTimeTS.eps}
    \caption{Simulation results for the distal compartment for continuous inflow (first plot on each row) or fluctuating inflow (i.e. `meals') for transit times of 1 d (top row) and 2 d (bottom row) and for 2 random seeds. Modelled pH varies with TSCFA and the RS fraction is 0.78. There are no bowel movements (i.e. outflow is continuous). See Table \ref{table:groups} for microbial groups.
    }
    \label{fig:flucInflowTt}
\end{figure}


\subsection*{Model Experiments}
We now use our model to simulate two scenarios -- firstly, the effects of decreasing total carbohydrate intake and secondly, the effects of changing carbohydrate composition (whilst keeping total intake fixed) on the microbial commnuity and associated SCFA production.
Comparing our simulations with data from human volunteer experiments is not straightforward since in order to run our model, ingested food must be translated to non-digestable substrates reaching the colon. This is problematic due to unknown water consumption and transit times and uncertainties associated with the absorption rates of the ingested carbohydrate and protein higher up the digestive tract.
Thus we do not attempt to reproduce human experiments exactly but rather we run simulations based on variations to our standard model set up which are qualitatively similar, and then compare our results with the trends in the available data.


\subsubsection*{Effects of total dietary carbohydrate}
\label{sect:protein}
In this model experiment we investigate the effects of decreasing carbohydrate on the microbial community. 
Here we compare our results qualtitatively with the human dietary study of \cite{duncan} which explored the impacts of carefully controlled decreases in carbohydrate intake upon weight loss and microbial fermentation products in obese subjects using 3 diets -- a maintenance (M) diet, a high protein, moderate carbohydrate diet (HPMC) and a high protein, low carbohydrate diet (HPLC) (see Fig. \ref{fig:Invivo} for details).
This is of course, the composition for ingested food, which is not easily translated into substrate concentrations entering the colon. 
However, we can look at the general trends in SCFA and microbial composition with changing colonic carbohydrate intake rate.
Thus, in these model experiments we keep protein inflow to the colon at 10 g d$^{-1}$ (our default value) and then increase inflowing non-digestable (ND) carbohydrate from 10 g d$^{-1}$ to 60 g d$^{-1}$ in 10 g d$^{-1}$ intervals. 
To include the effects of different ND-carbohydrate composition we run the model for an resistant starch (RS) fraction of either 0.2 or 0.78 (the default value), with non-starch polysaccharides (NSP) making up the remaining ND-carbohydrate in each case. 
Although subject to large uncertainties, we estimate the RS fractions for the \cite{duncan} experiments of 0-0.6 (M diet), 0-0.68 (HPMC) and 0-0.12 (HPLC) (based on RS is 0--20\% of ingested starch \citep{capuano} and bio-available NSP is 75\% of ingested NSP \citep{slavin}).
Due to the low fibre nature of many of these simulations we run the model with a slightly longer transit time of 1.5 d and for both continuous inflow and meals.


Fig. \ref{fig:resultsDunc} shows the SCFA results from our model experiment and Fig. \ref{fig:Invivo} shows the results from the in vivo experiment. 
It is clear, from both the model and in vivo results that the proportion of butyrate increases as the amount of ND-carbohydrate in the diet increases. Furthermore, both model and in vivo results show an increase in TSCFA with ND-carbohydrate intake rate. 
Since \cite{duncan} also look at the relationship between butyrate concentration and grams of carbohydrate eaten per day, we plot butyrate against carbohydrate entering the colon (Fig. \ref{fig:ButyrateVsCforPexpt}) to compare with their Fig. 1. 
In both cases, butyrate concentration increases with incoming carbohydrate.
Furthermore, as seen in both the model and the data, the percentage of butyrate increases with carbohydrate intake (Fig. \ref{fig:ButyrateVsCforPexpt}). 
Analysis of 10 human studies involving 163 subjects has shown a highly significant increase in percentage butyrate with increasing total SCFA concentration in faecal samples \citep{labouyer}.


In terms of microbial composition, Fig. \ref{fig:resultsDunc} shows the results from our simulations are reasonably consistent across inflow type (meals or continuous), with B dominating at low carbohydrate intake.
When the RS fraction is low (i.e. when ND-carbohydrate is made up of 80\% NSP) then NBFD increase with increased C intake. 
Whereas when C is mostly RS then NBSD and BP1 increase with C. In both cases BP2 increase with increasing C intake.

\begin{figure}
    \centering
    \includegraphics[scale=0.4]{images/Table3.pdf}
    \includegraphics[scale=0.24]{images/InVivoExptsNew.eps}
    \caption{Table on left shows the dietary intake for two human studies (\cite{duncan} and \cite{walker2011}. PI, CI, SI and NSPI refer to ingested dietary protein, carbohydrate, starch and NSP. Note, starch value for the high RS diet in the \cite{walker2011} study included 26 g commercial RS. Bar plots show SCFA data from these studies. The bars in the plots have been ordered to show increasing RS fraction (estimated by SI/CI) for the Walker study (for comparison with Fig. \ref{fig:modelledRSfrac}) and increasing carbohydrate for the Duncan study (for comparison with Fig. \ref{fig:resultsDunc}).
    }
    \label{fig:Invivo}
\end{figure}


\begin{figure}
    \centering
    \includegraphics[scale=0.4]{images/modelledDuncanExpt.eps}
    \caption{Simulated Biomass and SCFA results for increasing carbohydrate inflow. Simulations are run with continuous substrate inflow (cts) and with `meals' for a transit time of 1.5 days. The results are the average over the last 3 weeks of a 28 day simulation and `meals' is the average over 4 stochastically-generated simulations. 
    }
    \label{fig:resultsDunc}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[scale=0.3]{images/ButyrateCProteinExpt.eps}
    \caption{Plot of modelled butyrate, \%butyrate and TSCFA against grams of carbohydrate entering the colon each day. Data from \cite{duncan} is shown in magenta - due to uncertainties in converting ingested starch to RS entering the colon there are large error bars on the amount of C (g/d). Error bars show C estimated by the sum of 75\% of ingested NSP plus 0-20\% of ingested starch.
    }
    \label{fig:ButyrateVsCforPexpt}
\end{figure}


\subsubsection*{Effects of carbohydrate composition}
Here we use the model to simulate the effects of changing carbohydrate composition on the microbial community composition by changing the ratio of RS to NSP whilst keeping the same amount of total incoming carbohydrate.
Fig. \ref{fig:modelledRSfrac} show a summary of the model results. 
Although there are differences between the continuous inflow/meals, and also for the different transit times (1 d and 3 d), the modelled trends are generally similar, showing a significant shift in community as the fraction of RS increases, an increase in TSCFA and changes in the SCFA ratios. 
We compare our results with a human dietary study (\cite{walker2011}, \cite{salonen} and references therein) examining the impact of switching the major type of ND-carbohydrate from wheat bran (NSP) to resistant starch.
Volunteers were provided successively with a maintenance diet, diets high in RS or NSPs and a reduced carbohydrate weight loss (WL) diet, over 10 weeks (Fig. \ref{fig:Invivo}).

There are large discrepancies between the SCFA predicted by our model (Fig. \ref{fig:modelledRSfrac}) and the measured SCFA data (Fig. \ref{fig:Invivo}). 
Our model predicts an increase in TSCFA as proportion of RS increases whereas total fecal SCFA were significantly lower for the RS and WL diets compared to the other two diets (in which NSP is higher). 
One possible explanation is that fermentation of RS occurs in more proximal regions of the colon compared with NSP fibre fermentation, such that there is greater absorption of the SCFA products.
A second possibility, also likely, is that transit times were longer for the RS diet than for the NSP diet, which we predict would result in decreased SCFA concentrations.
In our model the effect of the RS fraction on TSCFA is greater than the effect of transit time so we do not see this in Fig. \ref{fig:modelledRSfrac}.


The human study also included detailed compositional analysis of the fecal microbiota (\cite{walker2011}, \cite{salonen}) that revealed specific responses mainly by different groups of Firmicutes bacteria to the RS and NSP diets. This information was particularly important for the phylogenetic assignments to the functional groups used here and previously in the model of \cite{Kettle2015}. Our modelling predicts striking shifts in the microbial community, especially involving the NBSD, NBFD and butyrate-producing groups, with changing proportions of RS and NSP fibre (Fig. \ref{fig:modelledRSfrac}). 
We should also note that in the volunteer experiments many bacterial species were not significantly altered by the RS-NSP switch in vivo \citep{walker2011} indicating that many may be generalists, able to switch quickly between energy sources.


\begin{figure}
    \centering
    \includegraphics[scale=0.5]{images/modelledRSfrac.eps}
    \caption{Biomass and SCFA results for changing the RS fraction of inflowing carbohydrate with continuous substrate inflow (`cts inflow') and with `meals'. Protein and carbohydrate inflow are 10 and 50 g d$^{-1}$ respectively. The results are the average over the last 3 weeks of a 28 day simulation and `meals' is the average over 4 stochastically-generated simulations. 
    }
    \label{fig:modelledRSfrac}
\end{figure}







\section*{Discussion}
The development of a complex model of the microbial community in the human colon, whose simulations compare well with data, represents a significant step forward. 
Previous models have been based on simpler microbial models (e.g. \cite{Cremer17}, \cite{Munoz}, \cite{moorthy}), or have not shown such a good agreement with data (e.g. \cite{smith}).
Our previous complex model community consisted of 10 functional groups, but the model was designed only to simulate continuous culture conditions in a chemostat \citep{Kettle2015}. 
Translating this 10-group model into an in vivo setting has required introducing multiple gut compartments, and the absorption of water and SCFA, followed by comparison with generally observed characteristics of the system. 
We were then able to use this model to examine the predicted impact of changes in the amount and type of non-digestible carbohydrate (fibre) present in human diets upon concentrations of fermentation products (SCFA) in different gut compartments and in stool. At the same time, we predict the likely impact of dietary changes and variations in gut transit upon microbiota composition and fermentation products. 
The model must be regarded as work in progress particularly with respect to microbiota composition. Predictions can however become improved and refined as more information becomes available in time.

Assignments of microbial taxa to our ten functional groups were based initially on evidence from cultured isolates. These assignments have since been supported and greatly extended by analysis of genes diagnostic for different fermentation pathways within genomes and metagenomes \citep{reichardt} and by molecular detection of species enriched within the community by defined growth substrates in chemostat experiments \citep{duncan16} and dietary intervention studies \citep{salonen}. Nevertheless, these assignments inevitably remain provisional and incomplete and we do not claim that the model predictions can be made precise at a phylogenetic level. More emphasis is placed in our model on the prediction of metabolic outputs based on microbial transformations and interactions. While there is relatively little phylogenetic overlap for example between producers of propionate and butyrate (\cite{reichardt}, \cite{louis17}) there are many cases where individual species are known to use multiple alternative substrates as energy sources, which complicates assignments. For this reason, more weight was given to fermentation pathways than to substrate preferences in defining the functional groups.  
However, it would also be possible to define completely different groupings that relate to other outputs (e.g. bile acid metabolism, or vitamin/micronutrient supply) in order to address specific questions.
Furthermore, it may well be worthwhile to increase the number of functional groups in the future. The large ‘B’ group for example currently includes members of the Bacteroidetes phylum, but its characteristics are mainly based on well-studied members of the \textit{Bacteroides} genus. We know that \textit{Prevotella} is another highly abundant genus of \textit{Bacteroidetes} in the human colon, but the two genera tend not to co-occur at high levels in the same individuals (\cite{wu}, \cite{chung}). Less is known about human colonic \textit{Prevotella}, for which there are relatively few cultured representatives, making it premature to create a separate grouping, but this would clearly be desirable in the future as their prevalence is reported to affect health and responses to dietary intervention.
In future it should become possible to define the relative abundance of functional groups (MFGs) and their relationship to phylogeny directly from genomic and metagenome analysis, by examining genes diagnostic for particular pathways and functions (e.g. Reichardt et al, 2014).

The parameter values for the microbial groups used in our model are from the intrinsic data frames in the microPop package (the only changes are to LactateProducers). 
Although the work presented here did not attempt to fit particular parameters to data, as we focussed on expanding the scope of the model (i.e. changing the environment from fermentor to colon), these values are easy to alter, e.g. \cite{wang} changed many of these parameters to achieve a better model fit to their data.
As well as adjusting the parameters for each group to represent inter-individual variation, groups can also be easily added or removed from the model through the input argument `microbeNames'.
Furthermore, it is also possible to include any number of strains (with varying parameter values) within each functional group in order to add more variation in outcome (see \cite{Kettle2015}) but we did not do this here in the interests of computational time.
It should also be noted that the parameter values are highly uncertain in many cases and within each of our functional groups there will be large variability due to adaptation and evolution. 
Given this, we do not claim that the model response is necessarily representative of what may happen in an individual's gut, rather it can be used as an aid to gain insight into the relative importance of the different processes we are currently aware of and potentially to highlight, those we are not. 

In addition to this, it must be noted that the default diet chosen here with 10 g of protein and 50 g of carbohydrate fibre reaching the colon each day could be revised for any given population.
However, converting from quantities of ingested food to substrate inflow to the colon is highly uncertain with large variations between studies, as well as technical issues with measuring this accurately. 
With more time, it would be interesting to investigate a larger range of typical diets but this was beyond the scope of the current work.


In summary, although performing reasonably well, the model has the potential to be considerably improved simply by altering the parameter values and existing settings, however, more fundamental changes such as those listed below could also be investigated in future work:
\begin{itemize}
\item Adding more functional groups or pathway switches in the existing functional groups. For example at present only the Bacteroides group can utilise protein but it is now known that some butyrate producers can also utilise amino acids \citep{louis17}
\item Our pH relation with TSCFA is very simplistic and could potentially be improved, although host secretions mean this is not necessarily straightforward.
\item Currently we set the transit time for the colon and then this is split between the 3 model compartments based on their relative sizes. An interesting addition would be to alter transit time based on the composition of the various substrates entering the colon. For example, increasing residence time for high protein and/or low fibre diets. Due to variation in individual response this may need to include significant uncertainty ranges.
\item Related to this is changing the absorption rate of water through the gut wall based on the diet, for example more water could remain in the gut on a high fibre diet.
\item A longer term goal would be to model the processes in the gastrointestinal tract preceding the colon in order to simulate how substrates entering the colon relate to dietary intake. This would allow more accurate prediction of microbial metabolite production based on diet.

\end{itemize} 


To conclude, our model helps to explain some important, but poorly understood, relationships that have been reported in human studies, including the increase in butyrate proportion with increasing total faecal SCFA \citep{labouyer}. This phenomenon has important implications in view of the claimed benefits of butyrate supply for colorectal cancer prevention and the health of the colonic mucosa (\cite{louis2014}, \cite{hamer2008}). 
The model also predicts increasing total faecal SCFA with greater fibre intake and more rapid gut transit.
Gut transit is also shown to have potentially important consequences for microbiota composition and gut metabolism. In addition, the model confirms that the amount and type of non-digestible carbohydrate in the diet has the potential to cause major changes in microbiota composition. 
The nature of such changes is, however, predicted to be influenced by patterns of meal feeding and by any effects of dietary components (e.g. dietary fibre) upon gut transit. 
Human studies suggest that they will also depend on the initial microbiota composition. 
There is potential to use the model to explore how the presence of particular functional groups (such as lactate-utilizers (Wang et al 2020)) within an individual's microbiota can influence their gut metabolism and response to dietary intervention. This may indeed be one of the most intriguing and fruitful applications of such modelling approaches in the future.

\section*{Materials and methods}

\subsection*{Software}
To facilitate continued research and future model development by other researchers we provide all model code on github (https://github.com/HelenKettle/microPopGutCode). The R package microPopGut is contained in the file microPopGut\_1.0.tar.gz. This can be downloaded and installed in R using install.packages(`microPopGut\_1.0.tar.gz'). Furthermore instructions on how to use the package are given in the supplementary file `gettingStartedWithMicroPopGut.pdf'.


\subsection*{Microbial Model}
The microbial functional group model is based on the model described by \cite{Kettle2015} and implemented using the R package microPop \citep{Kettle2018}.
The microbial groups include producers of the three major SCFA detected in fecal samples (acetate, butyrate and propionate) together with utilizers of acetate, lactate, succinate, formate and hydrogen (see Table \ref{table:groups} for a summary, or refer to \cite{Kettle2015} for more detail).
The model and its equations are described in detail by \cite{Kettle2015} and \cite{Kettle2018} so only a brief overview is given here
The microbial groups are defined as data frames within the R package and these are shown in section 3 of the Supp. Info..
Although this application uses the microbial parameters (e.g. maximum growth rates, yields etc) that are in the package's intrinsic data frames, these can be easily changed by either modifying the dataframe in R or by providing a new dataframe - either as an input csv file or by creating one in R. One of the input arguments to the function \textit{microPopGut()} is \textit{microbeNames} which allows the user to also enter other microbial groups.    


The growth substrates available in the large intestine are divided into four categories: protein (P), non starch polysaccharides (NSP), resistant starch (RS) and sugars (and oligosaccharides and sugar alcohols); for simplicity, all carbohydrate units are regarded as being hexoses. NSP comprise major components of dietary fibre including the structural polysaccharides of the plant cell wall (cellulose, xylan, pectin), whereas RS refers to the fraction of dietary starch that resists digestion in the small intestine.
We consider 10 major metabolites that arise from substrate fermentation: acetate, propionate, butyrate, lactate, succinate, formate, hydrogen, carbon dioxide, methane and ethanol. Six of these metabolites (acetate, lactate, succinate, formate, hydrogen and carbon dioxide) are also considered as substrates, because they are known to be consumed by some groups (cross-feeding). 
It is well known that pH affects growth rate therefore each group is assigned a preferred range of pH within which it can reach its maximum growth rate, but outside of which, its growth is reduced or zero.
We model the rate of bacterial growth using Monod kinetics and assume that from 1 g of resource, Y g of biomass is produced. We assume that resource that is taken up by microbes, but not used to produce biomass, is converted to metabolites. If not all of the resource is converted to biomass or to the metabolites represented in our model, it is discarded. This applies, for example, to many diverse fermentation products of proteins (e.g. phenols, amines) that are not among the 10 major products covered by the model. 
Although the model was initially developed to be run with multiple strains within each functional group, in the current work we do not do this due to the high CPU time associated with multiple compartments.


\subsection*{Inflow to colon}
\subsubsection*{Incoming substrates and water}
The main sources of nutrient for microbiota in the colon are complex dietary carbohydrates that are not absorbed higher up the digestive tract. 
We use a default value of 50 g d$^{-1}$ of carbohydrate, C, in our model and we vary the proportion of this which is NSP or RS using the RS fraction (i.e. RS/(RS+NSP) where RS+NSP=C). 
Based on \cite{Cremer17} and references therein, about 15 g of bio-available NSP and 30-40 g of RS enter the colon per day which gives us an RS fraction of 0.67-0.9 with average value of 0.78 which we use as our default value.
According to \cite{yao} less is known regarding dietary proteins, P, that escape digestion to reach the large intestine, although it is estimated that around 6 - 18 g P reaches the large intestine daily, the majority from the diet and a small proportion from endogenous origins.
Given this, here we assume that 10 g d$^{-1}$ of undigested P reaches the colon from dietary intake along with a small amount from mucin degradation (approx. 1  g d$^{-1}$).
\cite{Phillips} state that water enters at approximately 1.5 l d$^{-1}$  and about 90\% of this is absorbed by the colon.
\cite{StephenCummings} states that normal fecal daily output in Britain is 100-200 g d$^{-1}$ of which 25-50 g d$^{-1}$ is solid matter and the rest (50-175 g d$^{-1}$) is water. Thus if 90\% is absorbed then this indicates water inflow in the range 0.5 - 1.75 l d$^{-1}$. 
The midpoint of this range is 110 g d$^{-1}$ of water outflow which, if 90\% is absorbed, implies that the inflow of water is approximately 1100 g d$^{-1}$. This will clearly vary depending on the host's oral water intake but we use 1100 g d$^{-1}$ as our default value.
The default inflow values are summarised in Table \ref{table:drivers}.

\begin{table}
    \centering
    \caption{Summary of default values used in the model. Parameter values for the microbial groups are given in the Supp. Info. (section 3)}
\small
    \begin{tabular}{|l|l|l|}
        \hline
        \textbf{Symbol} & \textbf{Description} & \textbf{Default Value}  \\  \hline 
        $T_t$ & transit time through colon & 1.25 d \\%& 0.5-5   \\
        $\dot{P}_{diet}$ & protein inflow rate &  10 g d$^{-1}$\\% & 1-20 \\
        $\dot{C}_{diet}$ & carbohydrate inflow rate &  50 g d$^{-1}$ \\%& 20-80  \\
        $\dot{W}_{diet}$ & water inflow rate &  1100 g d$^{-1}$ \\%& 500-1750  \\
        $\dot{M}$ & mucin inflow rate &  5 g d$^{-1}$ \\%& 2.5-7.5  \\
        $K_{M}$ & half saturation constant for Mucin breakdown & 0.5 g l$^{-1}$  \\
        $a_{W}$ & rate of water absorption by host & 3  d$^{-1}$  \\
        $a_{Z}$ & rate of SCFA absorption by host & 9.6  d$^{-1}$\\
   \hline 
    \end{tabular}
    \label{table:drivers}
\end{table}   

\subsubsection*{Meals}
The normal human diet does not consist of continuous fixed inflow of substrate; for a more realistic substrate inflow to the colon we simulate eating 3 meals a day with randomly varying composition. We then approximate the passage of these meals through the stomach and small intestine to obtain a smoothed time series for substrate entering the colon. 
Note that we are not simulating all the food ingested by the host (most of which will not reach the colon) but rather simply trying to produce a more realistic time series for the substrates that we know reach the colon.

We specify three meals per day each with a duration of 30 minutes. This time-series is then passed through a one-compartment ordinary differential equation model representing the time spent in the stomach and small intestine (estimated to take 7 hours), i.e.
\begin{equation}
    \frac{ds(t)}{dt}=\dot{s(t)}_{in}-vs(t)
\end{equation}
where $v$=3.4 d$^{-1}$ (inverse of 7 h transit time in days); $\dot{s(t)}_{in}$ is time series representing 3 meals a day (g d$^{-1}$) and $t$ is time in days. The inflow to the colon (i.e. the outflow from small instestine) is given by $vs(t)$. 
The composition (in terms of P, NSP, RS and water (W)) of these meals varies randomly around the mean of each component (Table \ref{table:drivers}) for each meal. 
To generate such random fluctuations we draw samples for each meal from a gamma distribution (since this is always above zero) defined by a scale parameter ($\gamma_s$) and the daily average inflow of the substrate (g d$^{-1}$). 
We assume the magnitude of the substrate fluctations are proportional to the mean value. 
Preliminary simulations showed that $\gamma_s$ equal to half the mean value of each substrate gave a good variation for P, RS and NSP, and for water variation we assumed $\gamma_s$ was one tenth of the incoming daily flow. 
The distributions and flow patterns are shown in Fig. \ref{fig:substrateInflow}.


\begin{figure}
    \centering
    \includegraphics[scale=0.25]{images/inflowGamma.eps}
    \includegraphics[scale=0.4]{images/inflowTS.eps}
    \includegraphics[scale=0.5]{images/inflowComp.eps}
    \caption{
a) Gamma distribution from which random values are drawn to generate the composition of each meal (note water is not shown due to the large difference in magnitude between water and dietary substrates). b) The substrate inflow time series to the proximal colon after passing through the small intestine. Examples shown for 3 stochastic simulations starting with different seeds. c) Barplots showing the composition of 6 meals over 2 days for 3 different stochastic simulations.}
    \label{fig:substrateInflow}
\end{figure}


\subsubsection*{Mucin}
There is a further input of protein and carbohydrate from the host via the breakdown of host-released mucin by many strains in the B group \citep{Racheev} and in our NBFD group \citep{crost}. It is estimated that 2.7-7.3 g d$^{-1}$ of mucin, denoted $\dot{M}$, is secreted into the colon \citep{florin}, therefore we take the midpoint value 5 g/d.  We assume our mucin degraders break down 1 g of mucin into 0.05 g sulphate, 0.2 g P and 0.75 g C, based on \cite{Sung}, but consider their yield on mucin to be negligible compared with growth on other substrates. 
We split C equally between NSP and RS - this arbitrary choice did not affect model results since C from mucin (3.75 g d$^{-1}$ maximum) is much less than dietary C (50 g d$^{-1}$), but this should be revised if considering very different dietary drivers.
Since the compartments of the colon are not equal-sized we assume that the rate of mucin entering the colon is divided through the model compartments proportional to their relative volumes.
We assume this enters the colon at a fixed, continuous rate and mucin-derived P and C are a function of the mass of mucin degraders, $D_M$ (B and NBFD), such that,
\begin{eqnarray}
    \dot{C(t)}&=&0.75\frac{D_M(t)}{D_M(t)+K_MW_v(t)}\dot{M}\\
    \dot{P(t)}&=&0.2\frac{D_M(t)}{D_M(t)+K_MW_v(t)}\dot{M}
\end{eqnarray}
where $C$, $P$, $D_M$ are in mass units and the over dot indicates a rate (e.g. g d$^{-1}$), $t$ is time  and $W_v$ is the volume of water in the model compartment. $K_M$ (g l$^{-1}$) is chosen such that if $D_M<<K_MW_v$ then there is minimal breakdown of mucin and if $D_M>>K_MW_v$ there is maximal breakdown. The smaller the value of $K_M$ the more breakdown there will be at low concentrations of mucin degraders. We set $K_M$=0.5 g l$^{-1}$ based on preliminary model simulations.


\subsection*{Absorption by host}
SCFA and water are both absorbed by the host through the gut wall; over 95\% of SCFA \citep{Topping} and approximately 90\% of incoming water is absorbed \citep{Phillips}.
Experiments by \cite{Ruppin} found that the absorption rates of SCFA to be approximately 0.4 h$^{-1}$ (i.e. 9.6  d$^{-1}$) with little difference in rates between the different SCFA (\cite{Ruppin}, \cite{Topping}).

We can estimate mathematically the specific water absorption rate required to give 90\% absorption of inflowing water for a given number of compartments in the colon ($N$) and a given transit time, $T_t$, using 
\begin{equation}
    a_W=\frac{16.95-9.72N+1.77N^2}{T_t}\label{eq:aW}
\end{equation}
(see Supp. Info. Section 1.3 for the derivation).
As a rough estimation, for a 3 compartment model with a transit time 1-1.5 days, gives $a_W\approx 3$  d$^{-1}$ (Supp. Info. Fig. S1a).
Given this will not be significantly affected by the microbial model (microbial uptake/production of water is small) this is a robust estimation.

To estimate the value of the specific absorption rate of SCFA, $a_Z$, we used a simple model (see Supp. Info. sections 1.1 and 1.4).
Estimating the value of the specific absorption rate of SCFA based on the values of SCFA given in the verification criteria and given our estimate for $a_W$ we found that it was necessary for the specific absorption rate to change along the colon (see Supp. Info. section 1.4). 
The best estimates were given by $a_Z$ values of 25.2, 4.2 and 9.2  d$^{-1}$ in the proximal, transverse and distal colon respectively. 
However, in the interests of a robust model (i.e. the fewer parameter values, the better) we made the decision to use one value for $a_Z$. Since the experimental value of 9.6  d$^{-1}$ compares well with our estimate in the distal colon we set $a_Z$=9.6  d$^{-1}$ throughout. It should be noted though that our model results could potentially be improved by varying $a_Z$ between model compartments.


\subsection*{pH}
Calculating pH in our model is not straightforward due to a lack of necessary state variables as well as pH buffering via secretions from the host. However, observations tell us the pH in the colon goes from 5.7 in the proximal, 6.2 in the transverse and 6.6 in the descending colon and TSCFA in these regions is around 123 mM, 117 mM and 80 mM respectively \citep{cummings}. Therefore an approximate approach is to simply make pH a function of TSCFA. 
Fitting a line through the above points gives us the following relationship
\begin{equation}
    \mbox{pH}=8.02-0.0174\times \mbox{TSCFA}.
\label{eq:pH(TSCFA)}
\end{equation}
which we further limit by setting the minimum and maximum pH values at 5 and 8 respectively i.e. if the TSCFA values give predicted pH outside of this range (Fig. \ref{fig:pHRelandTol}).

The impact of pH on microbial growth is modelled via a pH limitation function whereby there is a range over which there is no limit on growth but outside of this range the growth rate decreases linearly to reach zero at the specified outer limits. Thus there are 4 parameters used to describe the pH tolerance -- 2 for the inner range where there is no limit on growth and 2 for the outer range outside which there is no growth -- an example is shown in Fig. \ref{fig:pHRelandTol}.
The pH tolerance range for each microbial group is specified under the entry `pHcorners' in the data frame for each group and shown in Supp. Info. section 3.

\begin{figure}
    \centering
    \includegraphics[scale=0.3]{images/pHTSCFAandTol.eps}
    \caption{a) Relating pH to TSCFA using Eq. \ref{eq:pH(TSCFA)} and data from \citep{cummings}. b) Example of microbial tolerance to pH. 
A pH tolerance function of this form is specified individually for each microbial group in our model.  
}
    \label{fig:pHRelandTol}
\end{figure}


\subsection*{Fecal outflow}
Fecal outflow (g d$^{-1}$) at time, $t$, is given by $m_d(t)V_d$ where $m_d(t)$ is the mass in the distal colon (i.e. microbes, unconsumed substrate, microbial metabolites and water) and $V_d$ is the specific wash out rate from the colon (the inverse of the time spent in the distal colon).
For continuous outflow (as is used in most gut models) we compute the specific wash out rate from each compartment by assuming the fraction of time spent in compartment, $i$, is proportional to its volume fraction, thus 
\begin{equation}
    T_i=\frac{v_i}{v_{colon}}T_t
\end{equation}
where $v_i$ is the volume of compartment $i$ and $v_{colon}$ is the total volume of the colon. The specific wash out rate is then $V_i=1/T_i$.

If we introduce bowel movements then, assuming the distal colon is approximately emptied for each bowel movement, the total transit time is given by 
\begin{equation}
    T_t=\Sigma_{i=1}^2T_i + \frac{1}{N_{BM}}
\end{equation}
where $N_{BM}$ is the number of bowel movements per day.
For example, using volume measurements (Table \ref{fig:Validation}B) and assuming a total transit time of 1 day would mean about 45\% of the transit time is spent in the proximal and transverse colon and about 55\% of the day spent in the distal, which would be similar to 2 bowel movements per day. 
In model experiments where we vary the number of bowel movements per day we also change the time spent in the rest of the colon since we assume increased bowel movements are indicative of a general increase in passage rate.
We estimate the wash out rate from the colon during a bowel movement, $V_{BM}$, by
\begin{equation}
    V_{BM}=-\frac{\ln(f_d)}{\Delta t_{BM}}
\end{equation}
where $f_d$ is the fraction of mass left in the distal colon after the bowel movement and $\Delta t_{BM}$ is the time taken for the bowel movement (d). For example, if a bowel movement takes 10 minutes to remove 90\% of the contents of the distal colon then $V_{BM}$ is 332  d$^{-1}$. This is not affected by the number of bowel movements per day.

\section*{Author Contributions}
Kettle wrote the model code and led the writing of the manuscript. Flint contributed to writing the manuscript. Flint and Louis contributed to all aspects of microbiolgy and all authors contributed critically to the drafts and gave final approval for publication.

\section*{Acknowledgments}
We thank the Scottish Goverment's Rural and Environment Science and Analytical Services Division (RESAS) for funding this research. 

\subsection*{Data Accessibility}
All model code is on github (https://github.com/HelenKettle/microPopGutCode). The R package microPopGut is contained in the file microPopGut\_1.0.tar.gz. This can be downloaded and installed in R using install.packages(`microPopGut\_1.0.tar.gz'). Furthermore instructions on how to use the package are given in the supplementary file `getStartedWithMicroPopGut.pdf'. The model output for the simulations described in this manuscript are included on figshare in the file ModelRuns.tar.gz (https://doi.org/10.6084/m9.figshare.21094558.v1). The plotting code is provided in the github repository.

\section*{Funding Statement}
The Scottish Goverment's Rural and Environment Science and Analytical Services Division (RESAS) funded this research.

\bibliography{refs}

\end{document}
